homework/grayscale-image/compression.cpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+#include <algorithm>
+#include <cstdint>
+#include <iostream>
+#include <iterator>
+#include <utility>
+#include <cstddef>
+
+constexpr size_t height = 32;
+constexpr size_t width = 32;
+
+std::vector<std::pair<uint8_t,uint8_t>> compressGrayscale(const std::array<std::array<uint8_t,width>,height> & arrayTemp) {
+    std::pair<uint8_t, uint8_t> p_test;
+    std::vector<std::pair<uint8_t,uint8_t>> vec2;
+        std::all_of(arrayTemp.begin(), arrayTemp.end(), [&vec2, &p_test](std::array<uint8_t, width> a){
```

Comment:
Hmm strange usage of std::all_of it should not modify anything. Only check some condition :)